sympy/vector/parametricregion.py

```diff
@@ -0,0 +1,48 @@
+from sympy.core.basic import Basic
+
+class ParametricRegion(Basic):
+    """
+    Represents a parametric region in space.
+    """
+    def __init__(self, system, *args):
```

Comment:
Basic subclasses should use `__new__` rather than `__init__` and should access their data from `self.args`.

Reply:
Thanks, I noticed this. All subclasses of `Basic` in sympy use `__new__`.

What is the reasoning behind this? AFAIK, we should use `__new__` if we intend to return objects of different types based on some conditions.

Reply:
> return objects of different types based on some conditions.

This is what is often done in SymPy. For example, `Rational.__new__` may return an instance of `Integer` and `Integer.__new__` may return `S.One`.